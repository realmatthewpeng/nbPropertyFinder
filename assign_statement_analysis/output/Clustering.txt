Total number of lines found: 18

df["DayDiff"] = DataFrame([get_datetime(val) for val in df["Checkin Date"]]) - DataFrame([get_datetime(val) for val in df["Snapshot Date"]])
df["DiscountDiff"] = df["Original Price"] - df["Discount Price"]
df = pd.read_csv("hotels_data.csv")
df["DiscountPerc"] = (df["DiscountDiff"]/df["Original Price"]) * 100
df["WeekDay"] = DataFrame([get_datetime(val).weekday() for val in df["Checkin Date"]])
df["Hotel_Count"] = df.groupby('Hotel Name')['Hotel Name'].transform('count')
descending_hotels = df.sort_values(by=['Hotel_Count'],ascending=False).reset_index()
most_common_hotels = descending_hotels[descending_hotels['Hotel Name'].isin(df_hotels)]
most_common_hotels["Checkin_Count"] = most_common_hotels.groupby('Checkin Date')['Checkin Date'].transform('count')
most_checkins = descending_most_common_hotels[descending_most_common_hotels['Checkin Date'].isin(common_checkins_list)]
descending_most_common_hotels = most_common_hotels.sort_values(by=['Checkin_Count'],ascending=False).reset_index()
new_df =  DataFrame.from_records(combs,columns=["Hotel Name","Checkin Date","Discount Code","Discount Price"])
most_checkins = most_checkins.append(new_df)
vector = pd.DataFrame.from_records(rows)
hotels = hotels[[0,"cluster"]]
hotels = pd.DataFrame.from_records(vector.values)
hotels["Count"] = hotels.groupby("cluster")[0].transform("count")
hotels["cluster"] = clusters

Incorrect Lines:


Precision = 1.0
Recall = 0.6923076923076923

most_checkins["Discount Price"]= most_checkins.groupby(['Hotel Name','Checkin Date','Discount Code'])["Discount Price"].transform('min')
checkin_hotel_discount = most_checkins[["Hotel Name","Checkin Date","Discount Code","Discount Price"]].reset_index()
discount_filtered = checkin_hotel_discount[checkin_hotel_discount['Discount Price'] > -1 ]
discount_filtered_grouped = discount_filtered.groupby('Hotel Name')['Discount Price']
discount_filtered["Normal"] = discount_filtered_grouped.transform(normalize_data)
discount_synth = checkin_hotel_discount[checkin_hotel_discount['Discount Price'] == -1 ]
discount_synth["Normal"] = -1
normal_dataFrame = discount_synth.append(discount_filtered)
