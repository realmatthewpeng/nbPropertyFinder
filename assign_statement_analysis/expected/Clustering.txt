df["WeekDay"] = DataFrame([get_datetime(val).weekday() for val in df["Checkin Date"]])
df["DayDiff"] = DataFrame([get_datetime(val) for val in df["Checkin Date"]]) - DataFrame([get_datetime(val) for val in df["Snapshot Date"]])
df["DiscountDiff"] = df["Original Price"] - df["Discount Price"]
df = pd.read_csv("hotels_data.csv")
df["DiscountPerc"] = (df["DiscountDiff"]/df["Original Price"]) * 100
most_common_hotels = descending_hotels[descending_hotels['Hotel Name'].isin(df_hotels)]
df["Hotel_Count"] = df.groupby('Hotel Name')['Hotel Name'].transform('count')
descending_hotels = df.sort_values(by=['Hotel_Count'],ascending=False).reset_index()
most_common_hotels["Checkin_Count"] = most_common_hotels.groupby('Checkin Date')['Checkin Date'].transform('count')
most_checkins = descending_most_common_hotels[descending_most_common_hotels['Checkin Date'].isin(common_checkins_list)]
descending_most_common_hotels = most_common_hotels.sort_values(by=['Checkin_Count'],ascending=False).reset_index()
new_df =  DataFrame.from_records(combs,columns=["Hotel Name","Checkin Date","Discount Code","Discount Price"])
most_checkins = most_checkins.append(new_df)
most_checkins["Discount Price"]= most_checkins.groupby(['Hotel Name','Checkin Date','Discount Code'])["Discount Price"].transform('min')
checkin_hotel_discount = most_checkins[["Hotel Name","Checkin Date","Discount Code","Discount Price"]].reset_index()
diff = max(x) - min(x)
discount_synth = checkin_hotel_discount[checkin_hotel_discount['Discount Price'] == -1 ]
discount_filtered["Normal"] = discount_filtered_grouped.transform(normalize_data)
discount_synth["Normal"] = -1
discount_filtered_grouped = discount_filtered.groupby('Hotel Name')['Discount Price']
discount_filtered = checkin_hotel_discount[checkin_hotel_discount['Discount Price'] > -1 ]
normal_dataFrame = discount_synth.append(discount_filtered)
vector = pd.DataFrame.from_records(rows)
labels = vector.values[:,0]
data = vector.values[:,1:160]
shc = cluster.hierarchy
Z = shc.linkage(data, method='ward')
hotels["cluster"] = clusters
hotels = hotels[[0,"cluster"]]
clusters = cluster.fit_predict(vector.values[:,1:160])
hotels = pd.DataFrame.from_records(vector.values)
hotels["Count"] = hotels.groupby("cluster")[0].transform("count")
pca_2d = pca.transform(data)